pkg/config/spec_test.go

```diff
@@ -693,4 +699,30 @@ var _ = Describe("GetSourceSize", func() {
 		Expect(err).ToNot(HaveOccurred())
 		Expect(sizeAfter).To(Equal(sizeBefore))
 	})
+	It("Skips the kubernetes host dir when calculating the sizes if set", func() {
+		sizeBefore, err := config.GetSourceSize(conf, imageSource)
+		Expect(err).To(BeNil())
+		Expect(sizeBefore).ToNot(BeZero())
+
+		Expect(os.Mkdir(filepath.Join(tempDir, "host"), os.ModePerm)).ToNot(HaveOccurred())
+		Expect(createFileOfSizeInMB(filepath.Join(tempDir, "host", "what.txt"), 200)).ToNot(HaveOccurred())
+		// Set env var like the suc upgrade does
+		Expect(os.Setenv("HOST_DIR", filepath.Join(tempDir, "host"))).ToNot(HaveOccurred())
+
+		sizeAfter, err := config.GetSourceSize(conf, imageSource)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(sizeAfter).To(Equal(sizeBefore))
+	})
+	It("Counts the kubernetes host dir when calculating the sizes if not set", func() {
+		sizeBefore, err := config.GetSourceSize(conf, imageSource)
+		Expect(err).To(BeNil())
+		Expect(sizeBefore).ToNot(BeZero())
+
+		Expect(os.Mkdir(filepath.Join(tempDir, "host"), os.ModePerm)).ToNot(HaveOccurred())
+		Expect(createFileOfSizeInMB(filepath.Join(tempDir, "host", "what.txt"), 200)).ToNot(HaveOccurred())
+
+		sizeAfter, err := config.GetSourceSize(conf, imageSource)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(sizeAfter).ToNot(Equal(sizeBefore))
```

Comment:
maybe check for equality with the actual size to avoid false positives?

Reply:
I dont know how to calculate it beforehand as it seems that it doesnt exactly matches the 2 * 200Mb files so I did not want to add a random number here :D

I can still add it if you want?

Reply:
I think we add a hardcoded value to that for safety or something. I would just print the actual value and if that is consistent, I would check against it. Just to be sure we don't return a 0 or something and the test still passes.

Reply:
ok, done, just had to lok at how we calculate the size instead of being lazy :p